systest/multi-tenancy/basic_test.go

```diff
@@ -21,238 +22,454 @@ package main
 import (
 	"context"
 	"fmt"
-	"net/http"
-	"os"
-	"path/filepath"
-	"testing"
+	"strings"
 	"time"
 
 	"github.com/stretchr/testify/require"
 
-	"github.com/dgraph-io/dgo/v230"
 	"github.com/dgraph-io/dgo/v230/protos/api"
+	"github.com/dgraph-io/dgraph/dgraphtest"
 	"github.com/dgraph-io/dgraph/ee/acl"
-	"github.com/dgraph-io/dgraph/graphql/e2e/common"
-	"github.com/dgraph-io/dgraph/graphql/schema"
-	"github.com/dgraph-io/dgraph/testutil"
 	"github.com/dgraph-io/dgraph/x"
 )
 
-type inputTripletsCount struct {
-	lowerLimit int
-	upperLimit int
-}
-
-func prepare(t *testing.T) {
-	dc := testutil.DgClientWithLogin(t, "groot", "password", x.GalaxyNamespace)
-	require.NoError(t, dc.Alter(context.Background(), &api.Operation{DropAll: true}))
-}
+const (
+	aclQueryTimeout = 5 * time.Second
+)
 
-var timeout = 5 * time.Second
+func (suite *MultitenancyTestSuite) TestAclBasic() {
+	t := suite.T()
 
-// TODO(Ahsan): This is just a basic test, for the purpose of development. The functions used in
-// this file can me made common to the other acl tests as well. Needs some refactoring as well.
-func TestAclBasic(t *testing.T) {
-	prepare(t)
-	galaxyToken, err := testutil.Login(t,
-		&testutil.LoginParams{UserID: "groot", Passwd: "password", Namespace: x.GalaxyNamespace})
+	// Galaxy Login
+	hcli, err := suite.dc.HTTPClient()
+	require.NoError(t, err)
+	err = hcli.LoginIntoNamespace(dgraphtest.DefaultUser, dgraphtest.DefaultPassword, x.GalaxyNamespace)
+	require.NotNil(t, hcli.AccessJwt, "galaxy token is nil")
+	require.NoError(t, err, "login with namespace failed")
 
-	require.NotNil(t, galaxyToken, "galaxy token is nil")
-	require.NoError(t, err, "login failed")
 	// Create a new namespace
-	ns, err := testutil.CreateNamespaceWithRetry(t, galaxyToken)
+	ns, err := hcli.AddNamespace()
 	require.NoError(t, err)
 	require.Greater(t, int(ns), 0)
 
 	// Add some data to namespace 1
-	dc := testutil.DgClientWithLogin(t, "groot", "password", ns)
-	testutil.AddData(t, dc)
+	gcli, cleanup, err := suite.dc.Client()
+	defer cleanup()
+	require.NoError(t, err)
+	require.NoError(t, gcli.LoginIntoNamespace(context.Background(),
+		dgraphtest.DefaultUser, dgraphtest.DefaultPassword, ns))
+	suite.AddData(gcli)
 
-	query := `
-		{
-			me(func: has(name)) {
-				nickname
-				name
-			}
+	// Upgrade
+	suite.Upgrade()
```

Comment:
I feel a bit uneasy calling upgrade here, given that this would be used for integration tests also. Can we improve this?

Reply:
What doesn't feel right to you? For integration tests, this function is empty. It looks odd I agree, we couldn't think of a better way though.